stdlib/3/typing.pyi

```diff
@@ -178,7 +178,18 @@ class Container(Generic[_T_co]):
     @abstractmethod
     def __contains__(self, x: object) -> bool: ...
 
-class Sequence(Iterable[_T_co], Container[_T_co], Sized, Reversible[_T_co], Generic[_T_co]):
+if sys.version_info >= (3, 6):
+    class Collection(Sized, Iterable[_T_co], Container[_T_co], Generic[_T_co]):
+        pass
+
+if sys.version_info >= (3, 6):
+    class _Sequence(Collection[_T_co], Reversible[_T_co], Generic[_T_co]):
```

Comment:
I'm not so keen to have this intermediate class. But I can't think of a completely clean solution (without duplicating the entire rest of the class definition). Maybe we can do the following though?
```python
if sys.version_info >= (3, 6):
    class Collection(Sized, Iterable[_T_co], Container[_T_co], Generic[_T_co]): pass
    _Collection = Collection
else:
    class _Collection(Sized, Iterable[_T_co], Container[_T_co], Generic[_T_co]): pass
class Sequence(_Collection[_T_co], Generic[_T_co]):
    ...
```
(There's an even simpler solution, but it would make the official full name of the class `_Collection`.)